File: 2026-04-10-axiom/tether/src/compiler.js
import fs from 'fs';
import path from 'path';

function walkSync(dir, filelist = []) {
    let files;
    try {
        files = fs.readdirSync(dir);
    } catch (e) {
        return filelist;
    }

    for (const file of files) {
        if (file === 'node_modules' || file === '.git' || file === '.tetherrules') continue;
        const filepath = path.join(dir, file);
        try {
            if (fs.statSync(filepath).isDirectory()) {
                filelist = walkSync(filepath, filelist);
            } else {
                filelist.push(filepath);
            }
        } catch (e) {
            // Ignore unreadable files
        }
    }
    return filelist;
}

export function compileContext() {
    let contextBundle = '';
    const files = walkSync('./');
    for (const file of files) {
        try {
            const stats = fs.statSync(file);
            if (stats.size > 100 * 1024) continue; // skip files > 100KB
            const content = fs.readFileSync(file, 'utf8');
            // Basic binary check
            if (content.indexOf('\0') === -1) {
                contextBundle += `--- FILE: ${file} ---\n${content}\n`;
            }
        } catch (err) {
            console.error(`Compiler error: ${err.message}`);
        }
    }
    console.log(`Compiled ${contextBundle.length} bytes of context.`);
    return contextBundle;
}

File: 2026-04-10-axiom/tether-test/src/compiler.rs
use ignore::WalkBuilder;
use std::fs;

pub fn compile_context() -> String {
    let mut context_bundle = String::new();
    let walker = WalkBuilder::new("./")
        .hidden(false)
        .build();

    for result in walker {
        match result {
            Ok(entry) => {
                if !entry.file_type().map_or(false, |ft| ft.is_file()) {
                    continue;
                }
                
                let path = entry.path();
                if let Ok(content) = fs::read_to_string(path) {
                    context_bundle.push_str(&format!("--- FILE: {} ---\n{}\n", path.display(), content));
                }
            }
            Err(err) => tracing::error!("Compiler error: {}", err),
        }
    }
    
    tracing::info!("Compiled {} bytes of context.", context_bundle.len());
    context_bundle
}
